python/lsst/rubintv/production/rubinTv.py

```diff
@@ -825,3 +850,201 @@ def beat(self, ensure=False, customFlatlinePeriod=None):
         if (elapsed >= self.uploadPeriod) or ensure or customFlatlinePeriod:
             if self.uploader.uploadHeartbeat(self.handle, forecast):  # returns True on successful upload
                 self.lastUpload = now  # only reset this if the upload was successful
+
+
+class CalibrateCcdRunner():
+    """Class for running CharacterizeImageTask and CalibrateTasks on images.
+
+    Runs these tasks and writes shards with various measured quantities for
+    upload to the table.
+    """
+    def __init__(self, outputRoot, *, doRaise=False, embargo=False):
+        self.dataProduct = 'quickLookExp'
+        self.uploader = Uploader()
+        self.butler = makeDefaultLatissButler(embargo=embargo, writeable=True)
+        self.log = _LOG.getChild("monitorChannel")
+        self.channel = 'auxtel_characterizeImage'
+        self.watcher = Watcher(self.dataProduct, self.channel)
+        self.fig = plt.figure(figsize=(12, 12))
+        self.doRaise = doRaise
+        self.shardsDir = os.path.join(outputRoot, 'shards')
+        self.outputRunName = "LATISS/runs/quickLook/1"
+
+        config = CharacterizeImageConfig()
+        basicConfig = CharacterizeImageConfig()
+        obs_lsst = getPackageDir("obs_lsst")
+        config.load(os.path.join(obs_lsst, "config", "characterizeImage.py"))
+        config.load(os.path.join(obs_lsst, "config", "latiss", "characterizeImage.py"))
+        config.measurement = basicConfig.measurement
+
+        config.doApCorr = False
+        config.doDeblend = False
+        self.charImage = CharacterizeImageTask(config=config)
+
+        config = CalibrateConfig()
+        basicConfig = CalibrateConfig()
+        config.load(os.path.join(obs_lsst, "config", "calibrate.py"))
+        config.load(os.path.join(obs_lsst, "config", "latiss", "calibrate.py"))
+        config.measurement = basicConfig.measurement
+
+        config.doApCorr = False
+        config.doDeblend = False
+
+        self.calibrate = CalibrateTask(config=config, icSourceSchema=self.charImage.schema)
+
+    def _getRefObjLoader(self, refcatName, dataId, config):
+        """Get a reference object loader for a given refcat and dataId.
+        """
+        refs = self.butler.registry.queryDatasets(refcatName, dataId=dataId).expanded()
+        refs = list(refs)
```

Comment:
Given that this wrapping in `list()` is _critical_ for this to work consistently, I think it needs a comment to say so, lest generator-lover "optimize" this out in future.